NOTES-daneroo.md

```diff
@@ -0,0 +1,104 @@
+# Notes for PR(daneroo) feature/listing branch
+
+I don't expect this to be merged as-is. I mostly wanted to share my experiments and get feedback.
+
+If any of it is useful, I can craft proper commits with smaller features.
+
+*I am a first-time contributor, and not very experienced with Go, so would appreciate any feedback both on content and process.*
+
+## Overview
+
+When I first started to use the project, I experienced instability and would need to restart the download process a large number of times to get to the end. The process would timeout after 1000-4000 images. To test, I use two accounts one with ~700 images, and one with 31k images.
+
+The timeouts were mostly coming from the `navLeft()` iterator, so I disabled actual downloading and focused on traversal. (`-list` option). I also added an option (`-all`) to bypass the incremental behavior of `$dldir/.lastDone`. I then added `-vt (VerboseTiming`, to isolate timing metrics reporting specifically.
+
+The last main issue was the degradation of performance over time. Probably due to resource leaks in the webapp, especially when iterating at high speed. I found that the simplest solution to this was simply to force a reload every 1000 iterations. This typically takes about `1s` and avoids the `10X` slowdown I was observing.
+
+The following are details of each of the major changes.
+
+## `navLeft()`
+
+- By adding the `-list` option, its functionality is isolated.
+- Remove `WaitReady("body",..)`, as it has no effect (The document stays loaded when we navigate from image to image), and replace it with a loop (until `location != prevLocation`).
```

Comment:
I suspected that waiting on the body was useless, but was not sure, thanks.
However, I've never seen any problem with the left navigation itself so I fail to see why you added the loop around the location. Especially since that's what we already do in navN.
Can you please explain what happens for you in navLeft without this loop?

Reply:
It does not cause any issues, unless we skip the downloading.
It was an attempt to separate the concerns of navigating and downloading.